### PR TITLE
Showcasing useFetch custom React Hook for API data fetching

### PR DESCRIPTION
**Pull request by antoniosaleme**
## Description

This PR demonstrates a custom React Hook, useFetch, designed to simplify data fetching from an API. Although this code is intended as an example and not for merging into the master branch, it provides a clear illustration of how a custom hook can be implemented for fetching data from a given URL.

The useFetch hook leverages the useState and useEffect hooks from the React library. It maintains the state of the fetched data, loading status, and any errors that might occur. When the hook is called with a URL, it automatically initiates the data fetching process, updating the state accordingly. Once the data has been fetched, the state is updated to reflect the new data, loading status, and any errors.

The useFetch hook can be integrated into any React component that requires data fetching and state management, allowing for easy and efficient interaction with APIs.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## How has this PR been Tested?

- [x] Manually

### CODE REVIEW

File: client/src/pages/GalleryPage.jsx [hunk 1]
```diff
@@ -3,22 +3,24 @@ import axios from 'axios';
...
-      .then((response) => {
-        setGallery(response.data);
-      })
-      .catch((err) => console.log(err));
-  }, []);
+  const { data, isLoading, hasError } = useFetch('/api/gallery');
 
-  const filtered = gallery.filter((oneData) => {
+  if (isLoading) {
+    return <h1>{Loading}</h1>;
```
Comment (Pablo-Mdz) [inline client/src/pages/GalleryPage.jsx:16]: we can implement also  an infinit icon "loading..." to do it interactive

File: client/src/pages/TopicPage.jsx [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
   const [refreshTopics, setRefreshTopics] = useState(false);
   const [addTopic, setAddTopic] = useState(false);
   const { isLoggedIn, user } = useContext(AuthContext);
 
-  useEffect(() => {
-    axios
-      .get('/api/topic/list-topics')
-      .then((response) => {
-        setTopics(response.data.topics);
-        setRefreshTopics();
```
Comment (Pablo-Mdz) [inline client/src/pages/TopicPage.jsx:21]: this use effect is usefull also for setRefreshTopic, how do you set it with the new implementation?

Comment (antoniosaleme) [reply client/src/pages/TopicPage.jsx:21]: Now is fixed, please check it out! 🙏🏼

File: client/src/pages/GalleryPage.jsx (hunk 1 above)
Comment (antoniosaleme) [reply client/src/pages/GalleryPage.jsx:16]: I agree, I was using the one provided by ironlauncher. 

we can also use this one:

```
<div role="status">
    <svg aria-hidden="true" class="inline w-10 h-10 mr-2 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600" viewBox="0 0 100 101" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z" fill="currentColor"/>
        <path d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z" fill="currentFill"/>
    </svg>
    <span class="sr-only">Loading...</span>
</div>
```

**Pull request closed without merging**